src/renderer/sw_engine/tvgSwCommon.h

```diff
@@ -446,21 +446,29 @@ static inline uint32_t opBlendLighten(uint32_t s, uint32_t d, TVG_UNUSED uint8_t
 
 static inline uint32_t opBlendColorDodge(uint32_t s, uint32_t d, TVG_UNUSED uint8_t a)
 {
-    // d / (1 - s)
+    // - general formula: d / (1 - s)
+    // - with edge cases
+    // if (d == 0) return 0
+    // else if (s == 1) return 1
+    // else return min(1, d / (1 - s))
     s = 0xffffffff - s;
-    auto c1 = (C1(s) == 0) ? C1(d) : std::min(C1(d) * 255 / C1(s), 255);
-    auto c2 = (C2(s) == 0) ? C2(d) : std::min(C2(d) * 255 / C2(s), 255);
-    auto c3 = (C3(s) == 0) ? C3(d) : std::min(C3(d) * 255 / C3(s), 255);
+    auto c1 = C1(d) == 0 ? 0 : (C1(s) == 0 ? 255 : std::min(C1(d) * 255 / C1(s), 255));
```

Comment:
`C1(s) == 0 ? 255` does this necessary? Otherwise let's simplify the logic.

Reply:
without this we have division by 0: ```C1(d) * 255 / C1(s)```
and runtime error:
```
../src/renderer/sw_engine/tvgSwCommon.h:455:54: runtime error: division by zero
SUMMARY: UndefinedBehaviorSanitizer: undefined-behavior ../src/renderer/sw_engine/tvgSwCommon.h:455:54 in 
AddressSanitizer:DEADLYSIGNAL
```